meson_cpu/log/meson.build

```diff
@@ -0,0 +1,65 @@
+foreach cpu_req_var : [
+  'CPU_CONF_BASELINE', 'CPU_BASELINE_NAMES',
+  'CPU_CONF_DISPATCH', 'CPU_DISPATCH_NAMES',
+  'CPU_MTARGETS_LOG',
+]
+  if not is_variable(cpu_req_var)
+    error('Variable', cpu_req_var, 'is not exist. CPU dispatcher is not properly initialized.')
+  endif
+endforeach
+
+targets_info = {}
+targets_config = {}
+foreach config_name, mtargets_info : CPU_MTARGETS_LOG
+  foreach target_name, target_info : mtargets_info
+    if target_name not in targets_info
+      targets_info += {target_name: target_info}
+    endif
+    if target_name not in targets_config
+      targets_config += {target_name: []}
+    endif
+    targets_config += {
+      target_name: targets_config[target_name] + [config_name]
+    }
+  endforeach
+endforeach
+
+target_info_tpl = \
+'''
+  Target      : @0@
+  Enabled     : @1@
+  Flags       : @2@
+  Detect      : @3@
+  Defines     : @4@
+  Undefines   : @5@
+  Sources     : @6@
+'''
+generated_targets = ''
+foreach target_name, target_info : targets_info
+  generated_targets += target_info_tpl.format(
+    target_name,
+    ' '.join(target_info['features']),
+    ' '.join(target_info['args']),
+    ' '.join(target_info['detect']),
+    ' '.join(target_info['defines']),
+    ' '.join(target_info['undefines']),
+    '\n                '.join(targets_config[target_name])
+  )
+endforeach
+
+message(
```

Comment:
Since its okay to count on debug logging instead then we can just erase this build log and provide this information: through:https://github.com/mesonbuild/meson/blob/de7652ed3637f053a6facf422a642dc6aa2b904e/mesonbuild/modules/features/module.py#L517-L523

Its already provides reasons behind escaping each target, we can just append the information about each enabled target.

Reply:
I have separated the log into two parts, the first part which is flushed to stdout. e.g. on x86:
```Bash
CPU Optimiztion Options
  baseline:
    Requested : min
    Enabled   : SSE SSE2 SSE3
  dispatch:
    Requested : max -xop -fma4
    Enabled   : SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_KNL AVX512_KNM AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL AVX512_SPR
```
The second part which related to generate target became part meson log file(debug), e.g. on x86:
<details>
<summary>Debug: Generated CPU Targets:</summary>

```Bash
Debug: Generated CPU Targets: 
  Target      : baseline
  Enabled     : SSE SSE2 SSE3
  Flags       : -msse -msse2 -msse3
  Detect      : SSE SSE2 SSE3
  Defines     : SSE2 SSE SSE3
  Undefines   : 
  Sources     : _umath_tests.dispatch.h
                argfunc.dispatch.h
                loops_arithm_fp.dispatch.h
                loops_arithmetic.dispatch.h
                loops_comparison.dispatch.h
                loops_exponent_log.dispatch.h
                loops_hyperbolic.dispatch.h
                loops_logical.dispatch.h
                loops_minmax.dispatch.h
                loops_modulo.dispatch.h
                loops_trigonometric.dispatch.h
                loops_umath_fp.dispatch.h
                loops_unary.dispatch.h
                loops_unary_fp.dispatch.h
                loops_unary_fp_le.dispatch.h
                loops_unary_complex.dispatch.h
                loops_autovec.dispatch.h
                _simd.dispatch.h

  Target      : AVX2
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C AVX2
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mavx2
  Detect      : POPCNT AVX F16C AVX2
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C AVX2
  Undefines   : 
  Sources     : _umath_tests.dispatch.h
                argfunc.dispatch.h
                loops_arithmetic.dispatch.h
                loops_comparison.dispatch.h
                loops_logical.dispatch.h
                loops_minmax.dispatch.h
                loops_unary.dispatch.h
                loops_autovec.dispatch.h
                _simd.dispatch.h

  Target      : SSE41
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1
  Detect      : SSE SSE2 SSE3 SSSE3 SSE41
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41
  Undefines   : 
  Sources     : _umath_tests.dispatch.h
                loops_arithmetic.dispatch.h
                loops_unary_fp.dispatch.h
                loops_unary_fp_le.dispatch.h

  Target      : AVX512_SKX
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -mavx2 -mno-mmx -mavx512f -mavx512cd -mavx512vl -mavx512bw -mavx512dq
  Detect      : AVX512F AVX512CD AVX512VL AVX512BW AVX512DQ
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512F_REDUCE AVX512CD AVX512_SKX AVX512VL AVX512BW AVX512DQ AVX512BW_MASK AVX512DQ_MASK
  Undefines   : 
  Sources     : argfunc.dispatch.h
                simd_qsort.dispatch.h
                loops_arithmetic.dispatch.h
                loops_comparison.dispatch.h
                loops_exponent_log.dispatch.h
                loops_hyperbolic.dispatch.h
                loops_logical.dispatch.h
                loops_minmax.dispatch.h
                loops_umath_fp.dispatch.h
                loops_unary.dispatch.h
                _simd.dispatch.h

  Target      : SSE42
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2
  Detect      : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42
  Undefines   : 
  Sources     : argfunc.dispatch.h
                loops_comparison.dispatch.h
                _simd.dispatch.h

  Target      : AVX512_SPR
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL AVX512_SPR
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -mavx2 -mno-mmx -mavx512f -mavx512cd -mavx512vl -mavx512bw -mavx512dq -mavx512vnni -mavx512ifma -mavx512vbmi -mavx512vbmi2 -mavx512bitalg -mavx512vpopcntdq -mavx512fp16
  Detect      : AVX512F AVX512CD AVX512VL AVX512BW AVX512DQ AVX512VNNI AVX512IFMA AVX512VBMI AVX512VBMI2 AVX512BITALG AVX512VPOPCNTDQ AVX512FP16
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512F_REDUCE AVX512CD AVX512_SKX AVX512VL AVX512BW AVX512DQ AVX512BW_MASK AVX512DQ_MASK AVX512_CLX AVX512VNNI AVX512_CNL AVX512IFMA AVX512VBMI AVX512_ICL AVX512VBMI2 AVX512BITALG AVX512VPOPCNTDQ AVX512_SPR AVX512FP16
  Undefines   : 
  Sources     : simd_qsort_16bit.dispatch.h

  Target      : AVX512_ICL
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -mavx2 -mno-mmx -mavx512f -mavx512cd -mavx512vl -mavx512bw -mavx512dq -mavx512vnni -mavx512ifma -mavx512vbmi -mavx512vbmi2 -mavx512bitalg -mavx512vpopcntdq
  Detect      : AVX512F AVX512CD AVX512VL AVX512BW AVX512DQ AVX512VNNI AVX512IFMA AVX512VBMI AVX512VBMI2 AVX512BITALG AVX512VPOPCNTDQ
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512F_REDUCE AVX512CD AVX512_SKX AVX512VL AVX512BW AVX512DQ AVX512BW_MASK AVX512DQ_MASK AVX512_CLX AVX512VNNI AVX512_CNL AVX512IFMA AVX512VBMI AVX512_ICL AVX512VBMI2 AVX512BITALG AVX512VPOPCNTDQ
  Undefines   : 
  Sources     : simd_qsort_16bit.dispatch.h

  Target      : FMA3__AVX2
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -mavx2
  Detect      : POPCNT AVX F16C FMA3 AVX2
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2
  Undefines   : 
  Sources     : loops_arithm_fp.dispatch.h
                loops_exponent_log.dispatch.h
                loops_hyperbolic.dispatch.h
                loops_trigonometric.dispatch.h
                loops_unary_complex.dispatch.h
                _simd.dispatch.h

  Target      : AVX512F
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -mavx2 -mno-mmx -mavx512f
  Detect      : AVX512F
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512F_REDUCE
  Undefines   : 
  Sources     : loops_arithmetic.dispatch.h
                loops_comparison.dispatch.h
                loops_exponent_log.dispatch.h
                loops_trigonometric.dispatch.h
                loops_unary_complex.dispatch.h
                _simd.dispatch.h

  Target      : FMA3
  Enabled     : SSE SSE2 SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3
  Flags       : -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma
  Detect      : POPCNT AVX F16C FMA3
  Defines     : SSE2 SSE SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3
  Undefines   : 
  Sources     : _simd.dispatch.h

```
</details>

Reply:
I ended remove the whole debug log script and count only multi_targets() debug messages instead, it became much simpler now.